reference/7.1/Microsoft.PowerShell.Core/About/about_Functions_Advanced_Argument_Completion.md

```diff
@@ -0,0 +1,364 @@
+---
+ms.date: 07/22/2021
+description: Explains the various argument completion options available with advanced functions.
+Locale: en-US
+online version: https://docs.microsoft.com/powershell/module/microsoft.powershell.core/about/about_functions_advanced_argument_completion?view=powershell-7.1&WT.mc_id=ps-gethelp
+schema: 2.0.0
+title: About functions advanced argument completion
+---
+# about_Functions_Advanced_Argument_Completion
+
+## Short description
+Argument completion allows you to add tab expansion values to parameters of
+advanced functions. The available values are calculated at runtime when the
+user presses the <kbd>Tab</kbd> key after the parameter name. There are several
+ways to define an argument completer for a parameter.
+
+## Long description
+
+Advanced functions allow you create function parameters with special
+attributes. These attributes can be used to validate the user's input or
+provide argument completion. This article describe the different ways you can
+implement argument completion for advanced functions.
+
+Advanced functions use the `CmdletBinding` attribute to identify them as
+functions that act like cmdlets. For more information about this attribute, see
+[about_Functions_CmdletBindingAttribute](about_Functions_CmdletBindingAttribute.md).
+
+For more information about the other parameter attributes, see
+[about_Functions_Advanced_Parameters](about_Functions_Advanced_Parameters.md).
+
+## ValidateSet attribute
+
+The **ValidateSet** attribute specifies a set of valid values for a parameter
+or variable and enables tab completion. PowerShell generates an error if a
+parameter or variable value doesn't match a value in the set. In the following
+example, the value of the **Detail** parameter can only be Low, Average, or
+High.
+
+```powershell
+Param(
+    [Parameter(Mandatory=$true)]
+    [ValidateSet('Low', 'Average', 'High')]
+    [String[]]
+    $Detail
+)
+```
+
+In the following example, the value of the variable `$flavor` must be either
+**Chocolate**, **Strawberry**, or **Vanilla**. The `ValidateSet` attribute can
+be used on any variable, not just parameters.
+
+```powershell
+[ValidateSet('Chocolate', 'Strawberry', 'Vanilla')]
+[String]$flavor = 'Strawberry'
+```
+
+The validation occurs whenever that variable is assigned even within the
+script. For example, the following results in an error at runtime:
+
+```powershell
+Param(
+    [ValidateSet('hello', 'world')]
+    [String]$Message
+)
+
+$Message = 'bye'
+```
+
+Beginning in PowerShell 7.0, tab expansion was added for the values of
+`ValidateSet`. For example, if you were typing the following variable
+definition:
+
+```
+[ValidateSet('Chocolate', 'Strawberry', 'Vanilla')][String]$flavor = 'S<Tab>
+```
+
+When you hit the <kbd>Tab</kbd> key, you would get the following result:
+
+```
+[ValidateSet('Chocolate', 'Strawberry', 'Vanilla')][String]$flavor = 'Strawberry'
```

Comment:
This needs to be updated if using my suggestion above

Reply:
Fixed